lib/wprof/gen_prof.rb

```diff
@@ -0,0 +1,62 @@
+class GenProf
+  def initialize(event, rec_type)
+    @event = event
+    @db_runtime = WProf::Config.get_value(:db_runtime)
+    @async_report = WProf::Config.get_value(:async)
+    generate_profiling(rec_type)
+  end
+
+  def generate_profiling(rec_type)
+    common_params
+    case rec_type
```

Comment:
Personal opinion: not a big fan of case statements...What about using a hash instead:

```
profiling_params = { 
standard: :app_params,
service: :service_params,
custom: :custom_params
}

send(profiling_params[rec_type.to_sym]
```

Reply:
Sorry, but not. Case statements is better for code quality (it's so much clear to read) and its performance better than HASH and Send (Its true and it's verified!)